### PR TITLE
fix(SSA): don't use string literal if byte is "form feed" ('\f')

### LINKED ISSUE #8651: SSA Parser: '\u' is not a valid escape sequence.
### Aim

Discovered testing https://github.com/noir-lang/noir/pull/8647

### Expected Behavior

The SSA parser should parse the printed SSA.

### Bug

```console
❯ cargo test -p noir_ast_fuzzer --test parser
    Finished `test` profile [optimized + debuginfo] target(s) in 0.12s
     Running tests/parser.rs (target/debug/deps/parser-ff4db352fbc612fb)

running 1 test
test arb_ssa_roundtrip ... FAILED

failures:

---- arb_ssa_roundtrip stdout ----
global G_A: [u8; 1] = [226];
unconstrained fn main(a: [u8; 1], b: pub Field) -> pub bool {
    let mut ctx_limit = 25;
    (!func_2(a, a[0], (&mut ctx_limit)))
}
fn func_1(_ctx_limit: &mut u32) -> Field {
    (-(G_A[((G_A[0] as u32) % 1)] as Field))
}
unconstrained fn func_2(mut a: [u8; 1], mut b: u8, ctx_limit: &mut u32) -> bool {
    if ((*ctx_limit) == 0) {
        true
    } else {
        *ctx_limit = ((*ctx_limit) - 1);
        if (!func_2([2], a[0], ctx_limit)) {
            a = if (!(!true)) {
                for idx_c in (39 % 8) .. (b % 8) {
                    let mut d = false;
                    break;
                    d = if (!func_2(a, b, ctx_limit)) {
                        {
                            let mut idx_e = 0;
                            loop {
                                if (idx_e == 9) {
                                    break
                                } else {
                                    idx_e = (idx_e + 1);
                                    continue;
                                    if func_2(a, (a[0] * a[0]), ctx_limit) {
                                        {
                                            let mut idx_f = 0;
                                            while d {
                                                if (idx_f == 3) {
                                                    break
                                                } else {
                                                    idx_f = (idx_f + 1);
                                                    ()
                                                }
                                            }
                                        };
                                    };
                                }
                            }
                        };
                        d
                    } else {
                        (8727269082665342204 == (G_A[0] as u64))
                    };
                    let mut g = d;
                };
                [12]
            } else {
                a
            };
            true
        } else {
            true
        }
    }
}


thread 'arb_ssa_roundtrip' panicked at tooling/ast_fuzzer/tests/parser.rs:56:13:
Could not parse SSA after step 6 (Inlining): 
g0 = u8 226
g1 = make_array [u8 226] : [u8; 1]

brillig(inline) fn main f0 {
  b0(v2: [u8; 1], v3: Field):
    v4 = allocate -> &mut u32
    store u32 25 at v4
    inc_rc v2
    v7 = array_get v2, index u32 0 -> u8
    v9 = call f1(v2, v7, v4) -> u1
    v10 = not v9
    return v10
}
brillig(inline_always) fn func_2 f1 {
  b0(v2: [u8; 1], v3: u8, v4: &mut u32):
    v8 = allocate -> &mut [u8; 1]
    store v2 at v8
    v9 = allocate -> &mut u8
    store v3 at v9
    v10 = load v4 -> u32
    v12 = eq v10, u32 0
    jmpif v12 then: b1, else: b2
  b1():
    jmp b3(u1 1)
  b2():
    v13 = load v4 -> u32
    v15 = sub v13, u32 1
    store v15 at v4
    v17 = make_array [u8 2] : [u8; 1]
    v18 = load v8 -> [u8; 1]
    v19 = array_get v18, index u32 0 -> u8
    v21 = call f1(v17, v19, v4) -> u1
    v22 = not v21
    jmpif v22 then: b4, else: b5
  b3(v5: u1):
    return v5
  b4():
    v24 = load v9 -> u8
    v25 = truncate v24 to 3 bits, max_bit_size: 8
    jmp b6(u8 7)
  b5():
    jmp b7(u1 1)
  b6(v6: u8):
    v27 = lt v6, v25
    jmpif v27 then: b8, else: b9
  b7(v7: u1):
    jmp b3(v7)
  b8():
    v28 = allocate -> &mut u1
    store u1 0 at v28
    v30 = load v8 -> [u8; 1]
    inc_rc v30
    v31 = load v9 -> u8
    v32 = call f1(v30, v31, v4) -> u1
    jmp b9()
  b9():

    v34 = make_array b"\u{c}"
                     ^^^^
                     '\u' is not a valid escape sequence. Use '\' for a literal backslash character.

    store v34 at v8
    jmp b7(u1 1)
}

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

arbtest failed!
    Seed: 0x55ad804b00100000




failures:
    arb_ssa_roundtrip

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.64s

error: test failed, to rerun pass `-p noir_ast_fuzzer --test parser`

```

### To Reproduce

```shell
NOIR_ARBTEST_SEED=0x8e62ca0c00100000 cargo test -q -p noir_ast_fuzzer --test parser
```

### Workaround

None

### Workaround Description

_No response_

### Additional Context

_No response_

### Project Impact

None

### Blocker Context

_No response_

### Nargo Version

nargo version = 1.0.0-beta.6 noirc version = 1.0.0-beta.6+b2bfdb7bb0fc661bfb27da365f108cf131771295 (git version hash: a12bb6745e36e6ad54a5ded176c5f5b121f342d5, is dirty: true)

### NoirJS Version

_No response_

### Proving Backend Tooling & Version

_No response_

### Would you like to submit a PR for this Issue?

None

### Support Needs

_No response_

### PR DESCRIPTION
**Pull request by asterite**
# Description

## Problem

Resolves #8651

## Documentation

Check one:
- [x] No documentation needed.
- [ ] Documentation included in this PR.
- [ ] **[For Experimental Features]** Documentation to be submitted in a separate PR.

# PR Checklist

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Review by aakoshh — APPROVED** [review]
Nice, the mother of all edge cases 😄

File: compiler/noirc_evaluator/src/ssa/parser/tests.rs [hunk 1]
```diff
@@ -126,6 +126,19 @@ fn test_make_byte_slice_with_string_literal() {
     assert_ssa_roundtrip(src);
 }
 
+#[test]
+fn test_does_not_use_byte_array_literal_for_form_feed() {
+    // 12 is '\f', which isn't available in string literals (becuase in Rust it's the same)
```
Comment (TomAFrench) [inline compiler/noirc_evaluator/src/ssa/parser/tests.rs:131]:
```suggestion
    // 12 is '\f', which isn't available in string literals (because in Rust it's the same)
```
